src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -878,9 +878,9 @@ def __flatten_key_value(key, value, flattened_data, depth, separator):
             if key in flattened_data:
                 logger.debug(
                     "The key %s already exist, value has been overwritten.", key)
-            flattened_data[key] = str(value)
+            flattened_data[key] = value if isinstance(value, str) else json.dumps(value)  # Ensure boolean values are properly stringified.
```

Comment:
Is there any change in behavior if the value is a number or null?

Reply:
There is no change for behavior with a number, however, null values where being converted to the string "None". 
The `null` value is maintained in this case.

Reply:
Great! I'm surprised we didnt get any complaints about this. Thanks for fixing it!